src/array.jl

```diff
@@ -433,9 +434,15 @@ function copy!(dest::CategoricalArray{T, N}, dstart::Integer,
     dest
 end
 
-copy!(dest::CategoricalArray{T, N}, src::CategoricalArray{T, N}) where {T,N} =
+CA_OR_CA_VIEW = Union{CategoricalArray, SubArray{A,B,C,D} where
```

Comment:
Type names should be in CamelCase. `D` isn't needed. Actually, you should even be able to use `SubArray{<:Any, <:Any, <:CategoricalArray}`.